### PR TITLE
Add TravisCI & ESLint

### LINKED ISSUE #4: Add TravisCI To Repository
Some helpful tutorials:
[5 Steps to build NodeJS using Travis CI](https://medium.com/@LihauTan/5-steps-to-build-nodejs-using-travis-ci-fd7bcd030106)
[Building a JavaScript and Node.js project](https://docs.travis-ci.com/user/languages/javascript-with-nodejs/)

### LINKED ISSUE #5: Add ESLint to Repository
[ESLint](https://eslint.org/) helps us set rules for syntax in our code. Examples include enforving the use of single quotes, use camelcase for variable naming and more. Use [this guide](https://eslint.org/docs/user-guide/getting-started) to get started by adding it to the repository.

- If it asks, use a eslint file with JSON format
- The list of rules we can set are [here](https://eslint.org/docs/rules/)
- [example eslint JSON file](https://github.com/evanugarte/SmartPhotoCategorizer/blob/master/.eslintrc.json)

### PR DESCRIPTION
**Pull request by jerrylee17**
Resolves #4 
Resolves #5

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "no-empty": [
+            "error",
+            {
+                "allowEmptyCatch": true
+            }
+        ],
+        "no-extra-semi": "error",
+        "no-var": 0
+    }
+}
```
Comment (evanugarte) [inline .eslintrc.json]: keep newline at the bottom of the file.

File: package.json [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
     "node",
     "glitch",
     "express"
-  ]
-}
+  ],
+  "devDependencies": {
+    "eslint": "^6.8.0"
+  }
+}
```
Comment (evanugarte) [inline package.json]: keep newline here

File: src/commands/cct.js [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-		if (author.permissions.has("MANAGE_CHANNELS") || author.permissions.has("ADMINISTRATOR")){
+    if (author.permissions.has('MANAGE_CHANNELS') ||
+      author.permissions.has('ADMINISTRATOR')) {
       message.guild.createChannel(str);
-      message.channel.send(author + " created " + str);
+      message.channel.send(author + ' created ' + str);
     }
-	},
+  },
 };
```
Comment (evanugarte) [inline src/commands/cct.js]: go through your files and add back the newlines

**Comment by evanugarte** [comment]
Also make your PR titles and commit messages present tense

File: src/commands/cct.js [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
-      return
+  execute(message, args) {
+    if (args.join(' ') == '') {
+      message.channel.send('You need to give the name of the channel!');
+      return;
     }
     const author = message.member;
-    var str = args.join(" ");
+    var str = args.join(' ');
     //message.channel.send(author.haspermission("MANAGE_CHANNELS"))
```
Comment (evanugarte) [inline src/commands/cct.js]:
```suggestion
```

File: src/commands/help.js [hunk 5]
```diff
@@ -8,45 +8,37 @@ module.exports = {
...
-        for (var j = 0; j < tmp.length; j++) {
+      for (let i = 0; i < categories.length; i++) {
+        let tmp = commands.array();
+        for (let j = 0; j < tmp.length; j++) {
           if (tmp[j].category == categories[i]) {
-            dict[categories[i]] += tmp[j].name + ", "
+            dict[categories[i]] += tmp[j].name + ', ';
           }
         }
         //message.channel.send("c " + coms)
```
Comment (evanugarte) [inline src/commands/help.js]:
```suggestion
```

File: src/commands/help.js [hunk 6]
```diff
@@ -8,45 +8,37 @@ module.exports = {
...
+        dict[categories[i]] = dict[categories[i]]
+          .substring(0, dict[categories[i]].length - 2);
+        helpEmbed.addField(categories[i], dict[categories[i]]);
       }
 
       message.channel.send(helpEmbed);
 
 
-
     } else {
```
Comment (evanugarte) [inline src/commands/help.js]: can we remove this else block for now?

File: src/commands/help.js [hunk 7]
```diff
@@ -8,45 +8,37 @@ module.exports = {
   execute(message, args) {
     const { commands } = message.client;
     //list of categories
-    const categories = (commands.map(command => command.category)).filter((x, i, a) => a.indexOf(x) == i);
+    const categories = (commands.map(command => command.category))
+      .filter((x, i, a) => a.indexOf(x) == i);
```
Comment (evanugarte) [inline src/commands/help.js]: can you add better documentation up top for what this line is doing? `x` `i` and `a` are obscure names.

File: src/commands/help.js [hunk 8]
```diff
@@ -8,45 +8,37 @@ module.exports = {
   execute(message, args) {
     const { commands } = message.client;
     //list of categories
```
Comment (evanugarte) [inline src/commands/help.js]: add a better description like
```suggestion
    // Here, we iterate through the list of the bots available commands. We first
    // get all the categories, and then...
```

File: src/commands/help.js [hunk 9]
```diff
@@ -8,45 +8,37 @@ module.exports = {
...
-    const categories = (commands.map(command => command.category)).filter((x, i, a) => a.indexOf(x) == i);
+    const categories = (commands.map(command => command.category))
+      .filter((x, i, a) => a.indexOf(x) == i);
     //dictionary of categories
     var dict = {};
```
Comment (evanugarte) [inline src/commands/help.js]: idk what dict is. can you update the variable name and add comments?

File: src/commands/help.js [hunk 10]
```diff
@@ -8,45 +8,37 @@ module.exports = {
...
 
     if (!args.length) {
       //data.push('Here\'s a list of all my commands:');
```
Comment (evanugarte) [inline src/commands/help.js]:
```suggestion
```

File: src/commands/cct.js (hunk 4 above)
Comment (jerrylee17) [reply src/commands/cct.js]: Honestly all of these files need to be refactored. This should be a separate issue in my opinion.

**Comment by evanugarte** [comment]
@jerrylee17 we can merge this in. lets focus on a class for now

**Pull request merged**